packages/smart-account/src/SmartAccount.ts

```diff
@@ -561,13 +561,21 @@ class SmartAccount extends EventEmitter {
 
     if (updateImplTrx.data != '0x') {
       batchTrx.push(updateImplTrx, transactionDto.transaction)
-      response = this.sendTransactionBatch({ transactions: batchTrx })
+      response = this.sendTransactionBatch({
+        transactions: batchTrx,
+        paymasterServiceData: transactionDto.paymasterServiceData
```

Comment:
what if tomorrow instead of TransactionDto we just take ethers TransactionRequest as first argument and second could be paymasterService custom data

Reply:
You never know what is going to happen tomorrow; life is forever changing and unfolding in surprising ways.
But anyways, not making changes related to overall structure in this PR

Reply:
If method is dev facing and need to plan changes carefully. and we're actually going to accept ethers TransactionRequest. If we pass extra optional params (appended towards the end by looking what's in current published package) and try to rule out any possible breaking changes that'd be good.

Reply:
In the current scenario and in the new one let us say it is still called TransactionDto.
Right now first element is   `transactions:TransactionType` and second being `paymasterServiceData: PaymasterServiceDataType`
When we move to ethers, the first element can become ethersTransaction: TransactionRequest and second element being the same?

Reply:
ethers transactionRequest has customData field which can be passed Record < string, object >

Reply:
we can go ahead with this and test it out. Can you review once again and make sure anything new appended is optional and remains last then let's merge and test it.

Reply:
Okay so when will we be shifting to ethers transactionRequest?

Reply:
Okay will review again once

Reply:
Checked
It is optional and last param everywhere expect when engine is there, in that case engine is the last param